ui/packages/catalog-ui-search/src/main/webapp/react-component/filter/filter-input/filter-location-input.js

```diff
@@ -0,0 +1,239 @@
+  dmsEastDirection: Direction.East,
+  dmsWestDirection: Direction.East,
+  lat: undefined,
+  lon: undefined,
+  dmsLat: '',
+  dmsLon: '',
+  dmsLatDirection: Direction.North,
+  dmsLonDirection: Direction.East,
+  radius: 1,
+  bbox: undefined,
+  polygon: undefined,
+  hasKeyword: false,
+  usng: undefined,
+  usngbb: undefined,
+  utmUpsEasting: undefined,
+  utmUpsNorthing: undefined,
+  utmUpsZone: 1,
+  utmUpsHemisphere: 'Northern',
+  utmUpsUpperLeftEasting: undefined,
+  utmUpsUpperLeftNorthing: undefined,
+  utmUpsUpperLeftZone: 1,
+  utmUpsUpperLeftHemisphere: 'Northern',
+  utmUpsLowerRightEasting: undefined,
+  utmUpsLowerRightNorthing: undefined,
+  utmUpsLowerRightZone: 1,
+  utmUpsLowerRightHemisphere: 'Northern',
+  line: undefined,
+  lineWidth: 1,
+}
+
+const minimumBuffer = 0.000001
+
+class LocationInput extends React.Component {
+  locationModel
+  constructor(props) {
+    super(props)
+    this.locationModel = new LocationOldModel()
+    this.state = this.locationModel.toJSON()
+    this.deserialize()
+  }
+  setModelState() {
+    this.setState(this.locationModel.toJSON())
+    this.onChange()
+  }
+  componentWillMount() {
+    this.locationModel.on('change', this.setModelState, this)
+  }
+  componentDidMount() {
+    this.props.listenTo(
+      this.locationModel,
+      'change:mapNorth change:mapSouth change:mapEast change:mapWest',
+      this.locationModel.setLatLon
+    )
+    this.props.listenTo(this.locationModel, 'change', this.updateMap)
+    this.props.listenTo(this.locationModel, 'change:polygon', () => {
+      if (this.locationModel.get('mode') !== 'poly') {
+        wreqr.vent.trigger('search:polydisplay', this.locationModel)
+      }
+    })
+    this.props.listenTo(this.locationModel, 'change:mode', () => {
+      this.clearLocation()
+    })
+  }
+  componentWillUnmount() {
+    this.locationModel.off('change', this.setModelState)
+    wreqr.vent.trigger('search:drawend', this.locationModel)
+  }
+  updateMap = () => {
+    const mode = this.locationModel.get('mode')
+    if (mode !== undefined && store.get('content').get('drawing') !== true) {
+      wreqr.vent.trigger('search:' + mode + 'display', this.locationModel)
+    }
+  }
+  deserialize = () => {
+    const filter = this.props.value ? this.props.value : null
+    if (filter === null || filter.value == undefined) {
+      return
+    }
+
+    this.locationModel.set(filterToLocationOldModel(filter))
+
+    switch (filter.type) {
+      // these cases are for when the model matches the filter model
+      case 'DWITHIN':
+        if (CQLUtils.isPointRadiusFilter(filter)) {
+          wreqr.vent.trigger('search:circledisplay', this.locationModel)
+        } else if (CQLUtils.isPolygonFilter(filter)) {
+          wreqr.vent.trigger('search:polydisplay', this.locationModel)
+        } else {
+          wreqr.vent.trigger('search:linedisplay', this.locationModel)
+        }
+        break
+      case 'INTERSECTS':
```

Comment:
✏️ could make a map of `filter.type` to the corresponding `search:display` string for each filter type and simplify your switch case.